unit_test/solver/matrix/test_matrix.py

```diff
@@ -152,6 +131,29 @@ def test_solve_large(self) -> None:
         # assert
         self.assertEqual(results, [5.0] * (size + 1))
 
+    def test_solve_singular(self) -> None:
+        """Test the solving of the matrix object."""
+        # arrange
+        matrix = Matrix()
+        index = matrix.add_unknowns(2)
+        equation1 = EquationObject()
+        equation1.indices = np.array([index, index + 1])
```

Comment:
Check the formatting of equation.indices! You need to define the format in NDArray because now a float is expected for an integer value

Reply:
I do not follow what you mean let's discuss

Reply:
Made issue out of this #122